presto-main/src/main/java/io/prestosql/execution/buffer/OutputBufferMemoryManager.java

```diff
@@ -108,6 +108,12 @@ public synchronized void updateMemoryUsage(long bytesAdded)
         return bufferBlockedFuture;
     }
 
+    /**
+     * Allows task execution even if the buffer is full.
+     * <p>
+     * It is used when some query fails because the buffer is full, to complete
+     * the corresponding future.
```

Comment:
Can we explain better this description?

I couldn't understand exactly what the method is doing and what is its purpose, for example:
- Why this output buffer can't be full?
- What is this flag "blockOnFull" being set?

I believe that this information is important so I can understand why this method exists. Can we improve it?

Reply:
changed to 
```
* By default each task has an buffer to hold the output data,
     * if this buffer reaches the maximum quantity of bytes, the is blocked.
     * This method allows a task to finish its execution even if the buffer
     * size limit is reached.
```